noir-projects/noir-protocol-circuits/crates/reset-kernel-lib/src/reset/transient_data.nr

```diff
@@ -100,37 +107,43 @@ pub fn verify_squashed_transient_data_with_hint_indexes<let NUM_NOTE_HASHES: u32
     }
     assert_eq(nullifiers_kept + num_squashed, NUM_NULLIFIERS, "Wrong number of nullifiers removed");
 
-    let mut note_logs_kept = 0;
-    let mut note_logs_removed = 0;
+    /*
+     * Log Squashing
+     */
+    let mut logs_propagated = 0;
+    let mut logs_squashed = 0;
     for i in 0..NUM_LOGS {
-        let log = note_logs[i];
+        let log = logs[i];
         let hint_index = transient_or_propagated_note_hash_indexes_for_logs[i];
-        // - For non-empty logs, their note_hash_counter can't be 0. It must match the counter of a note hash, which is verified in private_call_data_validator.
-        //   Therefor, is_propagated will be false if index_hint incorrectly points to an empty expected_note_hash and fails the check inside the else statement.
-        // - For empty logs, we expect is_propagated to be true, making them part of the note_logs_kept.
-        //   index_hint might point them to the same empty note. This doesn't matter because we increment the value of
-        //   note_logs_kept and ensure that the correct number of empty logs are present in the expected_note_logs.
-        let is_propagated = expected_note_hashes[hint_index].counter() == log.note_hash_counter;
+
+        // Logs with 0 note_hash_counter must be propagated.
+        // Logs linked to a propagated note hash will also be propagated. In this case, the log's note_hash_counter points to a note hash in expected_note_hashes.
+        let is_propagated = (log.inner.note_hash_counter == 0)
+            | (expected_note_hashes[hint_index].counter() == log.inner.note_hash_counter);
+
         let expected_index = if is_propagated {
-            note_logs_kept
+            logs_propagated
         } else {
-            NUM_LOGS - note_logs_removed - 1
+            NUM_LOGS - logs_squashed - 1
         };
-        let expected_log = expected_note_logs[expected_index];
+
+        let expected_log = expected_logs[expected_index];
         if is_propagated {
-            assert_eq(expected_log, log, "Propagated note log does not match");
-            note_logs_kept += 1;
+            assert_eq(expected_log, log, "Propagated private log does not match");
```

Comment:
This is where most of the increased gate count of reset circuit comes from. Because we are now checking the entire log (18 fields) instead of one field of log hash.

Reply:
That's weird indeed! If it's 18 fields per log and 64 logs, that's 1152 checks, and the increase in constraint counts is 30k so around 30 gates per check sounds like a lot